addons/l10n_mx/models/account.py

```diff
@@ -0,0 +1,75 @@
+# coding: utf-8
+# Copyright 2016 Vauxoo (https://www.vauxoo.com) <info@vauxoo.com>
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+import re
+from odoo import models, api, fields, _
+
+
+class AccountAccount(models.Model):
+    _inherit = 'account.account'
+
+    @api.multi
+    def assign_account_tag(self):
+        """Based on account code will be assigned by default the corresponding tag by tag code, this method is designed
+        to be called from a server action in order to ensure in the background the tag is being assigned.
+
+        The server action should be created like this:
+
+        ```
+        if object.company_id.country_id.id == env.ref('base.mx').id:
+            object.assign_account_tag()
+        ```
+        """
+        for account in self:
+            tags = account.search_tags()
+            account.write({'tag_ids': [(4, tags)]})
+
+    @api.multi
+    def search_tags(self):
+        self.ensure_one()
+        account_tag_obj = self.env['account.account.tag']
+        if not self.code:
+            return account_tag_obj
+        if not self.is_valid_account_number():
+            return account_tag_obj
+        account = self.get_account_tuple()
+        tag_parent = account_tag_obj.search([
+            ('name', 'ilike', str(account[0])),
+            ('color', '=', 1)])
+        tag = account_tag_obj.search([
+            ('name', 'ilike', '%s.%s' % (account[0], account[1])),
```

Comment:
the tag code is:

COD.E NAME_OF_TAG because the tag do not have code information.

i.e: a tag is "102.02 Banks" then we are looking by the first lart of the tag that's why we are using ilike.

Is that ok for you? or do you think in another approach¿?